yarn-project/simulator/src/public/side_effect_trace.ts

```diff
@@ -535,7 +536,7 @@ function createPublicExecutionRequest(avmEnvironment: AvmExecutionEnvironment):
   const callContext = CallContext.from({
     msgSender: avmEnvironment.sender,
     contractAddress: avmEnvironment.address,
-    functionSelector: avmEnvironment.functionSelector,
+    functionSelector: FunctionSelector.empty(),
```

Comment:
Please add a comment/link to ticket. I'm assuming this functionselector has to eventually be removed as well.

Reply:
Absolutely, this is one of the thing I noticed. Namely, CallContext being currently shared for private and public land, we have to set empty selector. I think we ultimately a PublicCallContext and PrivateCallContext.